go/worker/client/worker.go

```diff
@@ -58,8 +59,13 @@ func (w *Worker) Start() error {
 
 	// Wait for all runtimes to be initialized.
 	go func() {
-		for _, rt := range w.runtimes {
+		for id, rt := range w.runtimes {
 			<-rt.Initialized()
+
+			// After the runtime is finished initializing, enable registration if applicable.
+			if rp, ok := w.runtimeRoleProvider[id]; ok {
+				rp.SetAvailable(w.commonWorker.GetRuntime(id).RegisterNodeRuntime)
```

Comment:
It would be still nice to check if runtime exits.

Reply:
Is it even possible in this situation for `GetRuntime` to fail?  It iterates over basically the same list of runtimes, which is used to create the local `w.runtimes` map in the beginning.

Reply:
Probably not, unless we in the future change something (e.g. support tmp runtimes).